packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -190,7 +190,16 @@ Future<void> _runBuild(
 
   // MSBuild sends all output to stdout, including build errors. This surfaces
   // known error patterns.
-  final RegExp errorMatcher = RegExp(r':\s*(?:warning|(?:fatal )?error).*?:');
+  final RegExp errorMatcher = RegExp(
+	<String>[
+      // Known error messages
+      r'(:\s*(?:warning|(?:fatal )?error).*?:)',
+      r'Error detected in pubspec\.yaml:',
+
+      // Known secondary error lines for pubspec.yaml
+      r'No file or variants found for asset:',
+    ].join('|'),
```

Comment:
Please create multiple regular expressions rather than joining together one super regex for readability and debugability.

Reply:
Currently `stdoutErrorMatcher` expects a `RegExp`. Hence the super regex.

```dart
result = await globals.processUtils.stream(
      <String>[
        cmakePath,
        '--build',
        buildDir.path,
        '--config',
        sentenceCase(buildModeName),
        if (install)
          ...<String>['--target', 'INSTALL'],
        if (globals.logger.isVerbose)
          '--verbose',
      ],
      environment: <String, String>{
        if (globals.logger.isVerbose)
          'VERBOSE_SCRIPT_LOGGING': 'true',
      },
      trace: true,
      stdoutErrorMatcher: errorMatcher, // <--- HERE
    );
```

Do you approve reformating `RegExp? stdoutErrorMatcher` to `List<RegExp>? stdoutErrorMatcher` in stram function?

```dart
@override
  Future<int> stream(
    List<String> cmd, {
    String? workingDirectory,
    bool allowReentrantFlutter = false,
    String prefix = '',
    bool trace = false,
    RegExp? filter,
    RegExp? stdoutErrorMatcher,  // <--- HERE
    StringConverter? mapFunction,
    Map<String, String>? environment,
  }) async {
```

That would allow for something like this:

```

Reply:
ahh, I see. This is fine (as is).